packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -359,32 +344,59 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
         .accepted;
 
     await for (final state in stateMachine.stream) {
-      switch (state.type) {
-        case HostedUiStateType.notConfigured:
-        case HostedUiStateType.configuring:
-        case HostedUiStateType.signingIn:
-        case HostedUiStateType.signingOut:
+      switch (state) {
+        case HostedUiNotConfigured _:
+        case HostedUiConfiguring _:
+        case HostedUiSigningIn _:
+        case HostedUiSigningOut _:
+        case HostedUiSignedOut _:
           continue;
```

Comment:
Hmm how do you mean?

Reply:
isn't it that they are empty cases and continue to next case?

Reply:
No, `continue` in this context only affects the `for` loop. You can use `continue` in a switch statement to get fallthrough behavior, but you have to use labels and it looks a bit different: https://dartpad.dev/?channel=master&id=2c21e4067ffce9d989ab51d13f746e7c. Otherwise, `continue` is not allowed in a top-level switch.